Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/release/NewReleaseInfoFragmentViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package de.rki.coronawarnapp.ui.release
+
+import android.content.Context
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.BuildConfig
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.flowOf
+
+class NewReleaseInfoFragmentViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    @AppContext private val context: Context
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+    val routeToScreen: SingleLiveEvent<NewReleaseInfoFragmentNavigationEvents> = SingleLiveEvent()
+
+    fun onNextButtonClick() {
+        routeToScreen.postValue(NewReleaseInfoFragmentNavigationEvents.NavigateToMainActivity)
+    }
+
+    val appVersion = flowOf(
+        context.getString(R.string.release_info_version_title).format(BuildConfig.VERSION_NAME)
```

Comment:
This is evaluated on construction so the the `.asLiveData(context = dispatcherProvider.Default)` does not make it async.

To make the string format not happen on the main thread, you could write:
```kotlin
flow {
 emit(context.getString(R.string.release_info_version_title).format(BuildConfig.VERSION_NAME))
}
```

there isn't a huge cost to the string formatting here though.
It would be a bit nicer to not use the app context for formatting strings.

@chiljamgossow  added an extension function you could use here, I think (see `LazyString.kt`):

```kotlin
flow {
    R.string.release_info_version_title.toResolvingString(BuildConfig.VERSION_NAME).let { emit(it) }
}
```

Reply:
I tried using it with flow, the way you described it here but the thing is, I was observing it from the fragment using observe2 which requires LiveData. I wasn't really trying to make this string format async in the first place. As you said, it's not something that would actually give a performance hit so I think it's ok to live it as is.